### PR TITLE
testing: go vet submodule compatibility

### LINKED ISSUE #1161: testing: go vet fails with nested sub-modules
#1151 surfaced a problem with how go vet handles nested sub-modules.

Fails:
```
➜  go vet ./cloudsql/...          
go: warning: "./cloudsql/..." matched no packages
no packages to vet
```

```
➜  go vet ./cloudsql/mysql/database-sql/...
go: directory cloudsql/mysql/database-sql is outside main module
```

Succeeds:
```
➜  cd ./cloudsql/mysql/database-sql 
➜  go vet ./...
```

This is reproducible in the /run directory by creating a structure like this:

```
run/
├── editor
│   ├── editor
│   │   ├── Dockerfile
│   │   ├── go.mod
│   │   ├── go.sum
│   │   ├── ...
│   ├── sentiment
│   │   ├── Dockerfile
│   │   ├── go.mod
│   │   └── main.go
│   ├── spellcheck
│   │   ├── Dockerfile
│   │   ├── go.mod
│   │   └── main.go
```

Possible Solutions:

* Adjust our script to descend into every sub-module before running `go vet ./...`.
* Detect "matched no packages" and suppress error (go vet uses the same error code for a failed test as a failed execution, so this requires string matching stderr)
* Denylist on certain directories
* Fix go tool? Not clear what that would mean so probably would take too long.

### PR DESCRIPTION
**Pull request by grayside**
Fixes #1161 

This change takes the following steps:

* For a given directory, check to see if there are go files not inside a go module. If so, run go vet as normal.
* For each go module inside the target directory, run go vet. No recursion allowed.

### CODE REVIEW

**Comment by tbpg** [comment]
Looks like vet failed:

```
+ for i in $(find $targetDir -name "go.mod")
++ dirname ./memorystore/redis/gae_standard_deployment/go.mod
+ mod=./memorystore/redis/gae_standard_deployment
+ echo 'Running '\''go vet'\'' in ./memorystore/redis/gae_standard_deployment'
+ pushd ./memorystore/redis/gae_standard_deployment
Running 'go vet' in ./memorystore/redis/gae_standard_deployment
+ go vet ./...
/tmpfs/src/github/golang-samples/memorystore/redis/gae_standard_deployment /tmpfs/src/github/golang-samples
go: warning: "./..." matched no packages
no packages to vet
```

**Comment by grayside** [comment]
Interesting failure state. `memorystore/redis/gae_standard_deployment` doesn't have any go files, but does have a go.mod.

Is this a bug or a feature that `go vet` found a problem? Should we delete the go.mod and go.sum file?

**Comment by tbpg** [comment]
I think a feature? Seems odd to have a `go.mod` without any go files. I don't think we need it for the sample.

**Comment by grayside** [comment]
From that standpoint, I'll go ahead and clean up trivial violations in this PR.

**Comment by grayside** [comment]
@ericschmidtatwork this PR is removing an unneeded go.mod and go.sum file from a memorystore `app.yaml` sample. If you need this code in place let me know.

**Comment by grayside** [comment]
All tests passing. go vet caught another WAI error in run/hello-broken.

File: testing/kokoro/system_tests.sh [hunk 1]
```diff
@@ -136,7 +136,32 @@ fi
 # Do the easy stuff before running tests. Fail fast!
 if [ $GOLANG_SAMPLES_GO_VET ]; then
   diff -u <(echo -n) <(gofmt -d -s .)
-  go vet $TARGET
+
+  # Remove the golang-only triple-dot suffix.
+  targetDir="${TARGET%/...}"
```
Comment (tbpg) [inline testing/kokoro/system_tests.sh]: I think the convention is all lower case with underscores to separate words.

**Pull request merged**